openrank/apiserver/models.py

```diff
@@ -1,126 +1,84 @@
+class Enrollment(models.Model):
+    user_id = models.ForeignKey(User)
     contest_id = models.ForeignKey(Contest,on_delete=models.CASCADE)
-    marksobtained = models.IntegerField()
-    userDetails = models.charField()
-
-class Role(models.Model):
-	role = models.charField()
-
-
-class Question(models.Model):
-
-	description = models.TextField()
-	total_marks = models.IntegerField()
-	tags = models.manyToMany(Tag)
-
-	 
+    marks_obtained = models.IntegerField()
+    user_details = models.CharField(max_length=100) #chesinaka already ni!!!
 
+class Tag(models.Model):
+    tag_description =   models.CharField()     
+	
+class Question(models.Model):                      
+    description = models.TextField()            
+    total_marks = models.IntegerField()
+    tags = models.ManyToManyField(Tag)               
 
 class ContestQuestion(models.Model):
-	question_id = models.ForeignKey(Question,on_delete=models.CASCADE)
-	contest_id = models.ForeignKey(Contest,on_delete=models.CASCADE)
-	total_marks  = models.IntegerField()
-	time_limit   = models.IntegerField()
-	memory_limit = models.IntegerField()  
-	variable_constraints = models.CharField(max_length=45)
-
+    question_id = models.ForeignKey(Question,on_delete=models.CASCADE)
+    contest_id = models.ForeignKey(Contest,on_delete=models.CASCADE)
+    total_marks  = models.IntegerField()
+    time_limit   = models.IntegerField()
+    memory_limit = models.IntegerField()  
+    variable_constraints = models.CharField(max_length=100)
 
 class Language(models.Model):
-	language = models.charField()
-
+    language = models.CharField(max_length = 100)  
 
 class QuestionLanguage(models.Model):
-	question_id = models.ForeignKey(Question,on_delete=models.CASCADE)
-	language_id = models.ForeignKey(Language,on_delete=models.CASCADE)
-	function_signature =  models.TextField()
-	function_parameters = models.charField()
-	starter_code =        models.charField()
-
+    question_id = models.ForeignKey(Question,on_delete=models.CASCADE)
+    language_id = models.ForeignKey(Language,on_delete=models.CASCADE)
+    function_signature =  models.TextField()
+    function_parameters = models.CharField(max_length=100)
+    starter_code = models.CharField(max_length=100)
 
 class Tag(models.Model):
-	tag_description =   models.charField()  #A charField
-	questions = models.manyToMany(Question)
-    
+    tag_description = models.CharField(max_length=100)     
+    questions = models.ManyToManyField(Question)
 
+class TestCaseBatch(models.Model):
+    marks = models.IntegerField()
+    title = models.CharField()	
 class TestCase(models.Model):
-	_input = models.charField()
-	output = models.charField()
-	testcaseBatch_id = models.ForeignKey(TestCaseBatch,on_delete=models.CASCADE)
-	question_id = models.ForeignKey(Question,on_delete = models.CASCADE)
-	is_hidden = models.BooleanField()
+    testcase_input = models.CharField(max_length=100)  
+    output = models.CharField(max_length=100)
```

Comment:
testcase_input and output must be TextField as we need to use multiline. CharField produces single line input in UI by django as far as I have read. Also the input and output can sometimes be more than 100